src/main/java/org/ow2/proactive/catalog/service/CatalogObjectService.java

```diff
@@ -277,31 +277,16 @@ public CatalogObjectMetadata updateObjectMetadata(String bucketName, String name
             throw new TagsIsNotValidException(tags.get(), "tags");
         }
 
-        //synchronize project name values
-        for (KeyValueLabelMetadataEntity metadata : catalogObjectRevisionEntity.getKeyValueMetadataList()) {
-            if (metadata.getKey().equals(PROJECT_NAME)) {
-                metadata.setValue(projectName.orElse(""));
-            }
+        if (!catalogObjectRevisionEntity.getProjectName().equals(projectName.orElse("")) ||
+            !catalogObjectRevisionEntity.getTags().equals(tags.orElse(""))) {
+            catalogObjectRevisionEntity = buildCatalogObjectRevisionEntity("The project name or/and tags metadata are updated",
+                                                                           username,
+                                                                           projectName.orElse(""),
+                                                                           tags.orElse(""),
```

Comment:
will the portal provide existing tags when updating metadata ?

If not, this change means that when a user tries to update only the projectName, the code will set at the same time  tags to "".

Reply:
Yes, the portal provides existing tags when updating metadata.